glass/fields.py

```diff
@@ -3,22 +3,20 @@
 from __future__ import annotations
 
 import math
-import typing
 import warnings
+from collections.abc import Sequence
 from itertools import combinations_with_replacement, product
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Any
 
 import healpy as hp
 import numpy as np
-import numpy.typing as npt
 from gaussiancl import gaussiancl
 from transformcl import cltovar
 
 from glass import grf
 
 if TYPE_CHECKING:
-    import collections.abc
-    from collections.abc import Callable, Iterable, Iterator, Sequence
+    from collections.abc import Callable, Generator, Iterable, Iterator, Sequence
     from typing import Any
```

Comment:
```suggestion
```

Reply:
Good spot, although seems like I can remove the one outside the `TYPE_CHECKING` block